debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/transforms/DecodeLogicalDecodingMessageContent.java

```diff
@@ -0,0 +1,198 @@
+
+    private BoundedConcurrentHashMap<Schema, Schema> logicalDecodingMessageContentSchemaCache;
+    private ObjectMapper objectMapper;
+    private JsonSchemaData jsonSchemaData;
+
+    @Override
+    public ConfigDef config() {
+        final ConfigDef config = new ConfigDef();
+        return config;
+    }
+
+    @Override
+    public void configure(final Map<String, ?> configs) {
+        final Configuration config = Configuration.from(configs);
+
+        objectMapper = new ObjectMapper();
+        CommonConnectorConfig.FieldNameAdjustmentMode fieldNameAdjustmentMode = CommonConnectorConfig.FieldNameAdjustmentMode.parse(
+                config.getString(CommonConnectorConfig.FIELD_NAME_ADJUSTMENT_MODE));
+        jsonSchemaData = new JsonSchemaData(EventRouterConfigDefinition.JsonPayloadNullFieldBehavior.IGNORE,
+                FieldNameSelector.defaultNonRelationalSelector(fieldNameAdjustmentMode.createAdjuster()));
+        logicalDecodingMessageContentSchemaCache = new BoundedConcurrentHashMap<>(10000, 10, BoundedConcurrentHashMap.Eviction.LRU);
+    }
+
+    @Override
+    public R apply(final R record) {
+        // ignore all messages that are not logical decoding messages
+        if (!Objects.equals(record.valueSchema().name(), POSTGRES_LOGICAL_DECODING_MESSAGE_SCHEMA_NAME)) {
+            LOGGER.debug("Ignore not a logical decoding message. Message key: \"{}\"", record.key());
+            return record;
+        }
+
+        Struct originalValue = requireStruct(record.value(), "Retrieve a record value");
+        Struct logicalDecodingMessageContent = getLogicalDecodingMessageContent(originalValue);
+        R recordWithoutMessageField = removeLogicalDecodingMessageContentField(record);
+
+        final Schema updatedValueSchema = getUpdatedValueSchema(logicalDecodingMessageContent.schema(), recordWithoutMessageField.valueSchema());
+        final Struct updatedValue = getUpdatedValue(updatedValueSchema, originalValue, logicalDecodingMessageContent);
+
+        return record.newRecord(
+                record.topic(),
+                record.kafkaPartition(),
+                null,
+                // clear prefix of a logical decoding message
+                null,
+                updatedValueSchema,
+                updatedValue,
+                record.timestamp(),
+                record.headers());
+    }
+
+    private Struct getLogicalDecodingMessageContent(Struct valueStruct) {
+        Struct logicalDecodingMessageStruct = requireStruct(valueStruct.get(DEBEZIUM_LOGICAL_DECODING_MESSAGE_KEY),
+                "Retrieve content of a logical decoding message");
+
+        if (logicalDecodingMessageStruct.schema().field(DEBEZIUM_LOGICAL_DECODING_MESSAGE_CONTENT_KEY).schema().type() != Schema.Type.BYTES) {
+            throw new DataException("The content of a logical decoding message is non-binary");
```

Comment:
Here and in other places - what is advantage of using `DataException` over `DebeziumException`?